staging/mausezahn.c

```diff
@@ -710,11 +719,20 @@ int getopts (int argc, char *argv[])
 				(ipv6_mode && get_ip6_range_dst(tx.ip_dst_txt, l)) || // returns 1 when no range has been specified
 				(!ipv6_mode && get_ip_range_dst(tx.ip_dst_txt)))
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4, 6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_dst = libnet_name2addr6 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE);
+					if (strncmp((char*)&tx.ip6_dst,(char*)&in6addr_error,sizeof(in6addr_error))==0) {
+					  fprintf(stderr, "Failed to set destination IPv6 address. Please check if destination is set to a valid IPv6 address.\n");
+					  return 1;
+					}
+				}
+				else {
+					if ((tx.ip_dst = libnet_name2addr4 (l, tx.ip_dst_txt, LIBNET_RESOLVE)) == -1){
+					  fprintf(stderr, "Failed to set destination IPv4 address. Please check if destination is set to a valid IPv4 address.\n");
+					  return 1;
+					}
+			  }
 			}
```

Comment:
So I have some thoughts, I think you can use the following alg (as a general approach):

    if (inet_pton(AF_INET, ...) == 1) {
        /* we have IPv4 address specified in X.X.X.X format */
    } else if (inet_pton(AF_INET6, ...) == 1) {
        /* we have IPv6 address in X:X::X format */
    } else if (libnet_name2addr4(...) == SUCCESS) {
        /* we have resolved IPv4 address for the hostname */
    } else ... similar

And I think it is enough.

Looking to the fiture I think if we will have:

    1) Implemented enough protos in trafgen_proto API
    2) Have generic trafgen proto generation API

then we can get rid of libnet if above's will cover all the libnet API required by mz.